adaptive_experimental/lib/navigation_rail.dart

```diff
@@ -164,12 +164,12 @@ class _RailItem extends StatelessWidget {
     );
   }
 
-  Animation _positionAnimation;
+  late Animation _positionAnimation;
 
   final Animation<double> animation;
   final NavigationRailKind labelKind;
   final bool selected;
-  final Icon icon;
+  final Widget icon;
```

Comment:
this doesn't require a change anywhere else?

Reply:
`icon` is only used as the child of `SizedBox` and a child of `Column`, so any `Widget` can satisfy. Moreover, the following code requires `icon` to be a `Widget`:

```dart
                icon: widget.currentIndex == i
                    ? widget.items[i].activeIcon
                    : widget.items[i].icon,
```

(Lines 128–130 on `develop` in file `navigation_rail.dart`)